sdk/communication/chat/src/main/java/com/azure/android/communication/chat/AzureCommunicationChatServiceAsyncClient.java

```diff
@@ -0,0 +1,490 @@
+     */
+    AzureCommunicationChatServiceAsyncClient(AzureCommunicationChatServiceImpl serviceClient) {
+        this.serviceClient = serviceClient;
+    }
+
+    /**
+     * Gets read receipts for a thread.
+     * 
+     * @param chatThreadId Thread id to get the read receipts for.
+     * @param callback the Callback that receives the response.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    public void listChatReadReceipts(String chatThreadId, final Callback<Page<ReadReceipt>> callback) {
+        this.serviceClient.listChatReadReceipts(chatThreadId, callback);
+    }
+
+    /**
+     * Gets read receipts for a thread.
+     * 
+     * @param chatThreadId Thread id to get the read receipts for.
+     * @param collectionCallback the Callback that receives the response collection.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    public void listChatReadReceiptsPages(String chatThreadId, final Callback<AsyncPagedDataCollection<ReadReceipt, Page<ReadReceipt>>> collectionCallback) {
+        this.serviceClient.listChatReadReceiptsPagesAsync(chatThreadId, collectionCallback);
+    }
+
+    /**
+     * Sends a read receipt event to a thread, on behalf of a user.
+     * 
+     * @param chatThreadId Thread id to send the read receipt event to.
+     * @param body Request payload for sending a read receipt.
+     * @param callback the Callback that receives the response.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    public void sendChatReadReceipt(String chatThreadId, SendReadReceiptRequest body, final Callback<Void> callback) {
+        this.serviceClient.sendChatReadReceipt(chatThreadId, body, callback);
+    }
+
+    /**
+     * Sends a message to a thread.
+     * 
+     * @param chatThreadId The thread id to send the message to.
+     * @param body Details of the message to send.
+     * @param callback the Callback that receives the response.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    public void sendChatMessage(String chatThreadId, SendChatMessageOptions body, final Callback<SendChatMessageResult> callback) {
```

Comment:
We may want to have a later talk about the guidelines on how to name these options bags for our public methods. I feel like it would be clearer for the user if this was called `sendChatMessageOptions` or something along those lines instead of `body`.

Also, I know that having all optional parameters in the options or not has been a contentious topic among our architects, so we might want to revisit this once there's a conclusion to that discussion.

These comments apply to all other places where we have options bags, also where they are named `body`.

Reply:
just to clarify: there is a discussion on "deriving" and generating Option model, by "deriving" I mean, a model not coming from swagger but we compute at codegen level, that's the one we want to eventually hold the non-required params. This is not done in codegen yet.

Here "SendChatMessageOptions" is a model coming from swagger not derived, its that swagger author / service named the model like that, which created the confusion.